### PR TITLE
Added BaseConstraint.default_violation_error_message attribute hint

### PR DESCRIPTION
**Pull request by RyanWalker277**
# Refs
#1418 

# Changes

Added `default_violation_error_message` in `BaseConstraint` stubs

### CODE REVIEW

File: django-stubs/db/models/constraints.pyi [hunk 1]
```diff
@@ -16,6 +16,7 @@ class Deferrable(Enum):
 class BaseConstraint:
     name: str
     violation_error_message: _StrOrPromise | None
+    default_violation_error_message: str = ...
```
Comment (intgr) [inline django-stubs/db/models/constraints.pyi]: This won't allow lazy translations. Lazy translations are somewhat tricky.

Note code in Django:

```
from django.utils.translation import gettext_lazy as _
...
class BaseConstraint:
    default_violation_error_message = _("Constraint “%(name)s” is violated.")
```

The `_(...)` or `gettext_lazy(...)` here returns a lazy-translated string, it's not actually an instance of `str`, we call it `_StrPromise`.

Typing it as `str` would prevent this assignment.

Most contexts that accept `_StrPromise` also accept `str`, so we use the `_StrOrPromise` type there (it's just an alias for `str | _StrPromise`).

This here should use `_StrOrPromise` as well.

Comment (intgr) [inline django-stubs/db/models/constraints.pyi]: We don't usually add `= ...` to class attributes -- that's only for function/method arguments.

```suggestion
    default_violation_error_message: str
```

**Review by intgr — CHANGES_REQUESTED** [review]
Thanks for picking this up

File: django-stubs/db/models/constraints.pyi (hunk 1 above)
Comment (RyanWalker277) [reply django-stubs/db/models/constraints.pyi]: Will keep this in mind in future changes.Thanks for explaining it so well!

Comment (RyanWalker277) [reply django-stubs/db/models/constraints.pyi]: Will keep this in mind, thanks for the review =)

**Review by intgr — APPROVED** [review]

**Pull request merged**